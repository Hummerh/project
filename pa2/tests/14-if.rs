fn main() {
      if (true) {
            if (false) {
                  1; 2; 3;
            };
            if (true) {
                  1; 2; 3;
            };
      };

      if (true) {
            if (false) {
                  1; 2; 3
            } else {
                  1; 2; 3
            }
      } else {
            1; 2; 3
      };
}
