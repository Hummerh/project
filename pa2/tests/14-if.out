(crate
    (items
        (fn-def
            (id
                (main))
            (block
                (if
                    (true)
                    (block
                        (if
                            (false)
                            (block
                                (lit-dec)
                                (lit-dec)
                                (lit-dec)))
                        (if
                            (true)
                            (block
                                (lit-dec)
                                (lit-dec)
                                (lit-dec)))))
                (if
                    (true)
                    (block
                        (if
                            (false)
                            (block
                                (lit-dec)
                                (lit-dec)
                                (lit-dec))
                            (block
                                (lit-dec)
                                (lit-dec)
                                (lit-dec))))
                    (block
                        (lit-dec)
                        (lit-dec)
                        (lit-dec)))))))